.github/workflows/encryption.yaml

```
@@ -0,0 +1,103 @@
+
+on:
+  pull_request: {}
+  push:
+    branches:
+      - master
+
+env:
+  KIND_VERSION: v0.9.0
+  KIND_CONFIG: .github/kind-config-encryption.yaml
+
+jobs:
+  installation-and-connectivitiy:
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set image tag
+        id: vars
+        run: |
+          if [ ${{ github.event.pull_request.head.sha }} != "" ]; then
+            echo ::set-output name=tag::${{ github.event.pull_request.head.sha }}
+          else
+            echo ::set-output name=tag::${{ github.sha }}
+          fi
+
+      - name: Install Cilium CLI
+        run: |
+          curl -LO https://github.com/cilium/cilium-cli/releases/download/v0.4/cilium-linux-amd64.tar.gz
+          sudo tar xzvfC cilium-linux-amd64.tar.gz /usr/bin
+          rm cilium-linux-amd64.tar.gz
+
+      - name: Create kind cluster
+        uses: helm/kind-action@v1.0.0
+        with:
+          version: ${{ env.KIND_VERSION }}
+          config: ${{ env.KIND_CONFIG }}
+
+      - name: Wait for images to be available
+        timeout-minutes: 10
+        shell: bash
+        run: |
+          until curl --silent -f -lSL "https://quay.io/api/v1/repository/${{ github.repository_owner }}/cilium-ci/tag/${{ steps.vars.outputs.tag }}/images" &> /dev/null; do sleep 45s; done
+          until curl --silent -f -lSL "https://quay.io/api/v1/repository/${{ github.repository_owner }}/operator-ci/tag/${{ steps.vars.outputs.tag }}/images" &> /dev/null; do sleep 45s; done
+
+      - name: Install Cilium
+        run: |
+          cilium install --encryption \
+            --config monitor-aggregation=none \
+            --agent-image=quay.io/${{ github.repository_owner }}/cilium-ci \
+            --operator-image=quay.io/${{ github.repository_owner }}/operator-generic-ci \
+            --version=${{ steps.vars.outputs.tag }}
+
+      - name: Enable Relay
+        run: |
+          cilium hubble enable
+
+      - name: Status
+        run: |
+          cilium status --wait
+
+      - name: Relay Port Forward
+        run: |
+          kubectl port-forward -n kube-system deployment/hubble-relay 4245:4245&
+          sleep 5s
+
+      - name: Connectivity Test
+        run: |
+          cilium connectivity test
+      
+      - name: Rotate Keys
+        run: |
+          KEYID=$(kubectl get secret -n kube-system cilium-ipsec-keys -o yaml|grep keys: | awk '{print $2}' | base64 -d | awk '{print $1}')
+          if [[ $KEYID -gt 3 ]]; then KEYID=1; fi
+          data=$(echo "{\"stringData\":{\"keys\":\"$((($KEYID+1))) "rfc4106\(gcm\(aes\)\)" $(echo $(dd if=/dev/urandom count=20 bs=1 2> /dev/null| xxd -p -c 64)) 128\"}}")
+          kubectl patch secret -n kube-system cilium-ipsec-keys -p="${data}" -v=1
+
+      - name: Status
+        run: |
+          cilium status --wait
+
+      - name: Connectivity Test
+        run: |
+          cilium connectivity test
+
+      - name: Cleanup
```

nit: This doesn't seem to be cleaning up, seems more like:

```suggestion
      - name:  Capture cilium-sysdump
```

---

I've removed the CI test for now from this PR as it will be more involved to get it into a useful state.